tests/components/gogogate2/test_cover.py

```diff
@@ -0,0 +1,531 @@
+        datetime_mock.now.return_value = datetime.now() + timedelta(seconds=60.1)
+        component_data.data_update_coordinator.api.info.return_value = (
+            closed_door_response
+        )
+        await component_data.data_update_coordinator.async_refresh()
+        await hass.services.async_call(
+            HA_DOMAIN, "update_entity", service_data={"entity_id": "cover.door1"},
+        )
+        await hass.async_block_till_done()
+        assert hass.states.get("cover.door1").state == STATE_CLOSED
+
+
+async def test_availability(
+    hass: HomeAssistant, component_factory: ComponentFactory
+) -> None:
+    """Test open and close."""
+    closed_door_response = InfoResponse(
+        user="user1",
+        gogogatename="gogogatename0",
+        model="",
+        apiversion="",
+        remoteaccessenabled=False,
+        remoteaccess="abc123.blah.blah",
+        firmwareversion="",
+        apicode="",
+        door1=Door(
+            door_id=1,
+            permission=True,
+            name="Door1",
+            mode=DoorMode.GARAGE,
+            status=DoorStatus.CLOSED,
+            sensor=True,
+            sensorid=None,
+            camera=False,
+            events=2,
+            temperature=None,
+        ),
+        door2=Door(
+            door_id=2,
+            permission=True,
+            name=None,
+            mode=DoorMode.GARAGE,
+            status=DoorStatus.UNDEFINED,
+            sensor=True,
+            sensorid=None,
+            camera=False,
+            events=0,
+            temperature=None,
+        ),
+        door3=Door(
+            door_id=3,
+            permission=True,
+            name=None,
+            mode=DoorMode.GARAGE,
+            status=DoorStatus.UNDEFINED,
+            sensor=True,
+            sensorid=None,
+            camera=False,
+            events=0,
+            temperature=None,
+        ),
+        outputs=Outputs(output1=True, output2=False, output3=True),
+        network=Network(ip=""),
+        wifi=Wifi(SSID="", linkquality="", signal=""),
+    )
+
+    await component_factory.configure_component()
+    assert hass.states.get("cover.door1") is None
+
+    component_data = await component_factory.run_config_flow(
+        config_data={
+            CONF_IP_ADDRESS: "127.0.0.2",
+            CONF_USERNAME: "user0",
+            CONF_PASSWORD: "password0",
+        }
+    )
+    assert hass.states.get("cover.door1").state == STATE_OPEN
+
+    component_data.api.info.side_effect = Exception("Error")
+    await component_data.data_update_coordinator.async_refresh()
+    await hass.async_block_till_done()
+    assert hass.states.get("cover.door1").state == STATE_UNAVAILABLE
+
+    component_data.api.info.side_effect = None
+    component_data.api.info.return_value = closed_door_response
+    await component_data.data_update_coordinator.async_refresh()
+    await hass.async_block_till_done()
+    assert hass.states.get("cover.door1").state == STATE_CLOSED
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5535891)